src/safe_db/db.rs

```diff
@@ -68,4 +105,46 @@ impl DB {
         }
         self.edges = EdgeDB::new(edges)
     }
+
+    /// This method calculates how much of their own tokens a user can send to a given receiver.
+    /// The transfer limit is based on the trust relationship between the user and the receiver,
+    /// denoted by the trust_percentage argument.
+    /// If the receiver is an organization, the method simply returns the balance of the sender, i.e.,
+    /// the user can send all their tokens to an organization.
+    /// If the receiver is a regular user, the method calculates the transfer limit based on the balance
+    /// of the receiver and the trust percentage. The method scales down the receiver's balance based
+    /// on the trust percentage and compares it with the original balance to calculate the maximum
+    /// amount the user can send.
+    /// The maximum amount a user can send is the smaller of the user's balance and the difference between
+    /// the scaled receiver's balance and the balance calculated based on the trust percentage.
+    /// @returns how much of their own tokens a user can send to receiver.
+    fn trust_transfer_limit(&self
```

Comment:
I'd improve the function by accepting the general case of sending any token, not only sender's own tokens, like in here: https://github.com/CirclesUBI/pathfinder2/pull/43/files#diff-4d5dbf803354caf6408bea02c0b734f850ad3aaa5d2ffea457510a0a2958c654